native/src/routeSegmentResult.cpp

```diff
@@ -372,4 +373,20 @@ vector<SHARED_PTR<RouteSegmentResult>> RouteSegmentResult::getAttachedRoutes(int
 	}
 }
 
+SHARED_PTR<LatLon> convertPoint(SHARED_PTR<RouteDataObject> o, int ind){
+	return std::make_shared<LatLon>(get31LatitudeY(o->pointsY[ind]),  get31LongitudeX(o->pointsX[ind]));
```

Comment:
Better to add check if "SHARED_PTR<RouteDataObject> o" is not nullptr "if (o)"
Also to add check if (ind >= o->pointsX.size())
return make_shared<LatLon>(nullptr); by default